torch/tensor.py

```diff
@@ -418,17 +418,11 @@ def __rpow__(self, other):
 
     @_wrap_type_error_to_not_implemented
     def __floordiv__(self, other):
-        result = self / other
-        if result.dtype.is_floating_point:
-            result = result.trunc()
-        return result
+        return self.floor_divide(other)
```

Comment:
does this work for non-tensors?  (can you add a test)?

In particular, things like:
`3. // tensor`
`3 // tensor`
` tensor // 3.`
`tensor // 3`

Reply:
Added test for the scalar case. As commented below, I left the `3 // tensor` implementation as-is.